livekit-plugins/livekit-plugins-deepgram/livekit/plugins/deepgram/stt.py

```diff
@@ -311,6 +342,10 @@ def _process_stream_event(self, data: dict) -> None:
             self._event_queue.put_nowait(start_event)
             return
 
+        if data["type"] == "UtteranceEnd":
+            self._commit_all_final_events()
```

Comment:
Can u add a link that mentions the docs?
The event handling is becoming quite complex here and I'm not sure if UterranceEnd sufficient by himself.
e.g we're warning when there is no _final_events, maybe this will happen and this is expected

Reply:
Made some adjustments, lmk if it makes sense. This logic is definitely confusing - It could be worth writing a custom test just for deepgram because they have two "VAD" systems that work differently. One is the endpointing and it's pure DSP. Background noise will trigger it. the other is utterance end which is more NLP based.